lib/sdf-server/src/extract.rs

```diff
@@ -271,6 +271,32 @@ async fn tenancy_from_claim(
     Ok(Tenancy(dal::Tenancy::new(claim.workspace_pk)))
 }
 
+/// Use instead of [`axum::extract::Query`] when the query contains array params using "[]"
+/// notation.
+///
+/// Inspiration : https://dev.to/pongsakornsemsuwan/rust-axum-extracting-query-param-of-vec-4pdm
+pub struct QueryWithVecParams<T>(pub T);
+
+#[async_trait]
+impl<T> FromRequestParts<AppState> for QueryWithVecParams<T>
+where
+    T: serde::de::DeserializeOwned,
+{
+    type Rejection = (StatusCode, Json<serde_json::Value>);
+
+    async fn from_request_parts(
+        parts: &mut Parts,
+        _state: &AppState,
+    ) -> Result<Self, Self::Rejection> {
+        let query = parts
+            .uri
+            .query()
+            .ok_or(not_found_error("no query string found in uri"))?;
+        let deserialized = serde_qs::from_str(query).map_err(|err| internal_error(err))?;
```

Comment:
Isn't this a 422?

Reply:
I mean: shouldn't this produce a 422 and not a 500? If we fail to deserialize it's because the input was invalid

Reply:
Same with the not found. That's invalid input, not a resource not found

Reply:
Ah fair point. Will fix

Reply:
(or maybe 400: Bad Request)